mezzanine/template/loader_tags.py

```diff
@@ -68,7 +72,13 @@ def find_template(self, name, context, peeking=False):
         # other loaders like the ``cached`` template loader, unwind its
         # internal loaders and add those instead.
         loaders = []
-        for loader_name in settings.TEMPLATE_LOADERS:
+        loader_names = set(chain.from_iterable(
+            [template_engine.get('OPTIONS', {}).get('loaders', [])
+             for template_engine in settings.TEMPLATES]))
```

Comment:
I ran into a regression here using the cached template loader with its list of internal loaders specified as a list as recommended in the Django docs, e.g.:

```python
TEMPLATES = [{
    'BACKEND': 'django.template.backends.django.DjangoTemplates',
    'DIRS': [os.path.join(BASE_DIR, 'templates')],
    'OPTIONS': {
        'loaders': [
            ('django.template.loaders.cached.Loader', [
                'django.template.loaders.filesystem.Loader',
                'django.template.loaders.app_directories.Loader',
            ]),
        ],
    },
}]
```

Because `list` isn't hashable, the tuple containing it isn't hashable, so Python can't construct the set and throws an exception. Using a set here is a subtle bug anyway because it doesn't maintain the loader order, and I think unconditionally adding the filesystem and app dirs loaders isn't quite right either.

Could we just use [`Engine.template_loaders`](https://github.com/django/django/blob/master/django/template/engine.py#L105) here instead of re-parsing the settings?